scripts/benchmarks_ci.py

```diff
@@ -40,8 +40,85 @@
 import micro_benchmarks
 
 
+class DotNetPerformanceModes(Action):
+    '''
+    Default: (Currently Tiered)
+
+    NoTiering: (Formerly called full_opt)
+        COMPlus_TieredCompilation=0
+            This includes R2R code, useful for comparison against Default and
+            JitOnly for changes to R2R code or tiering.
+
+    JitOnly: Maybe could be called FullOptJitOnly to make it clear what it does
+        COMPlus_TieredCompilation=0
+        COMPlus_ReadyToRun=0
+            This is JIT-only, useful for comparison against Default and NoTier
+            for changes to R2R code or tiering.
+
+    MinOpt:
+        COMPlus_TieredCompilation=0
+        COMPlus_JITMinOpts=1
+            Uses minopt-JIT for methods that do not have pregenerated code,
+            useful for startup time comparisons in scenario benchmarks that
+            include a startup time measurement (probably not for
+            microbenchmarks), probably not useful for a PR.
+
+    For PRs it is recommended to kick off Default, NoTiering, and JitOnly modes
+    '''
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if values:
+            # Remove potentially set environments.
+            if 'COMPlus_TieredCompilation' in os.environ:
+                os.environ.pop('COMPlus_TieredCompilation')
+            if 'COMPlus_ReadyToRun' in os.environ:
+                os.environ.pop('COMPlus_ReadyToRun')
+            if 'COMPlus_JITMinOpts' in os.environ:
+                os.environ.pop('COMPlus_JITMinOpts')
+
+            # Configure .NET Runtime
+            if values == 'Default':
+                pass
+            elif values == 'NoTiering':
+                os.environ['COMPlus_TieredCompilation'] = '0'
+            elif values == 'JitOnly':
+                os.environ['COMPlus_TieredCompilation'] = '0'
+                os.environ['COMPlus_ReadyToRun'] = '0'
+            elif values == 'MinOpt':
+                os.environ['COMPlus_TieredCompilation'] = '0'
+                os.environ['COMPlus_JITMinOpts'] = '1'
+            else:
+                raise ArgumentTypeError(
+                    'Unknown mode: {}'.format(values)
+                )
+
+            setattr(namespace, self.dest, values)
+
+    @staticmethod
+    def modes() -> list:
+        '''Available .NET Performance modes.'''
+        return ['Default', 'NoTiering', 'JitOnly', 'MinOpt']
+
+    @staticmethod
+    def validate(usr_mode: str) -> str:
+        '''Default .NET performance mode.'''
+        requested_mode = None
+        for mode in DotNetPerformanceModes.modes():
+            if usr_mode.casefold() == mode.casefold():
+                requested_mode = mode
+                break
+        if not requested_mode:
+            raise ArgumentTypeError('Unknown mode: {}'.format(usr_mode))
```

Comment:
Would it make sense to default to the `Default` mode, or if making them explicit, to a mode appropriate for the runtime?

Reply:
Nevermind this is just validating a specified one